File: Exercise/Hard/02. Caesar Cipher Encryption/Code/Caesar Cipher Encryption.java
import java.util.Scanner;

public class CaesarCipher {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String input = scanner.nextLine();

        int shift = scanner.nextInt();

        String encryptedText = caesarCipher(input, shift);
        System.out.println(encryptedText);
    }

    public static String caesarCipher(String input, int shift) {
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < input.length(); i++) {
            char ch = input.charAt(i);

            if (Character.isLetter(ch)) {
                char base = Character.isUpperCase(ch) ? 'A' : 'a';
                ch = (char) ((ch - base + shift) % 26 + base);
            }

            result.append(ch);
        }

        return result.toString();
    }
}

File: Exercise/Hard/02. Caesar Cipher Encryption/Code/Caesar Cipher Encryption.js
function caesarCipher(input, shift) {
    let result = '';

    for (let i = 0; i < input.length; i++) {
        let char = input[i];

        if (char.match(/[a-z]/i)) {
            let code = input.charCodeAt(i);

            if (code >= 65 && code <= 90) {
                char = String.fromCharCode(((code - 65 + shift) % 26) + 65);
            } else if (code >= 97 && code <= 122) { 
                char = String.fromCharCode(((code - 97 + shift) % 26) + 97);
            }
        }

        result += char;
    }

    return result;
}

let input = "A friend in need is a friend indeed";
let shift = 20;
let encryptedText = caesarCipher(input, shift);
console.log(encryptedText);

